test-js/reportviewer-spec.js

```diff
@@ -12,170 +12,210 @@
-                reportViewer.reportPrompt.panel.paramDefn.paginate = true;
-                reportViewer.reportPrompt.panel.paramDefn.page = 5;
-                reportViewer.reportPrompt.panel.paramDefn.totalPages = 8;
-                reportViewer.view.updatePageControl();
-                expect(reportViewer.view._setAcceptedPage).toHaveBeenCalledWith("5");
-                expect(reportViewer.reportPrompt.api.operation.setParameterValue).toHaveBeenCalledWith("accepted-page", "5");
-              });
+            it("true", function() {
+              reportViewer.reportPrompt.panel.paramDefn.paginate = true;
+              reportViewer.reportPrompt.panel.paramDefn.page = 5;
+              reportViewer.reportPrompt.panel.paramDefn.totalPages = 8;
+              reportViewer.view.updatePageControl();
+              expect(reportViewer.view._setAcceptedPage).toHaveBeenCalledWith("5");
+              expect(reportViewer.reportPrompt.api.operation.setParameterValue).toHaveBeenCalledWith("accepted-page", "5");
             });
           });
+        });
+      });
 
-          describe("_buildReportContentOptions", function() {
+      describe("_buildReportContentOptions", function() {
+        it("should call the _buildReportContentOptions on the report viewer prompt", function() {
+          var options = {
+            "option1": "option1",
+            'action': 'testAction'
+          };
+          spyOn(reportViewer.reportPrompt, "_buildReportContentOptions").and.returnValue(options);
 
-            it("should call the _buildReportContentOptions on the report viewer prompt", function() {
-              var options = {"option1": "option1", 'action': 'testAction'};
-              spyOn(reportViewer.reportPrompt, "_buildReportContentOptions").and.returnValue(options);
+          var result = reportViewer._buildReportContentOptions();
 
-              var result = reportViewer._buildReportContentOptions();
+          expect(reportViewer.reportPrompt._buildReportContentOptions).toHaveBeenCalledWith('REPORT');
+          expect(result).toBe(options);
+          expect(result['name']).toBe(options['action']);
+        });
+      });
 
-              expect(reportViewer.reportPrompt._buildReportContentOptions).toHaveBeenCalledWith('REPORT');
+      describe("updateLayout", function() {
+        beforeEach(function() {
+          spyOn(reportViewer.view, "_hideToolbarPromptControls");
+          spyOn(reportViewer.view, "_showReportContent");
+          spyOn(reportViewer.view, "_initLayout");
+        });
 
-              expect(result).toBe(options);
-              expect(result['name']).toBe(options['action']);
-            });
-          });
+        // This test checks subscription on afterUpdate event.
+        it("should update layout after refreshing prompt panel", function(done) {
+          spyOn(reportViewer.view, "updateLayout");
+
+          reportViewer.reportPrompt.api.operation.refreshPrompt();
+          done();
```

Comment:
@annapopova1 since the expect is after the done is called, it won't actually count towards the success/failure of the test.
This needs to be revisited, it seems to be causing one other test to fail when using karma.ci.conf.js